src/components/CategoriesPills.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+import React from "react";
+import { View, StyleSheet, ScrollView, PixelRatio } from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+import Text from "./Text";
+import text from "../constants/text";
+import eventCategories from "../constants/event-categories";
+import locale from "../data/locale";
+
+import {
+  blackColor,
+  whiteColor,
+  darkBlueGreyTwoColor,
+  eucalyptusGreenColor
+} from "../constants/colors";
+
+const categoryStyleColor = (category: string) => {
+  const categoryData = eventCategories[locale][category];
+  return { color: categoryData.contrast ? blackColor : whiteColor };
+};
+
+const categoryStyleBackgroundColor = (category: string) => {
+  const categoryData = eventCategories[locale][category];
+  return { backgroundColor: categoryData.color };
+};
+
+type Props = {
+  selectedCategories: Set<string>,
+  style?: StyleObj
+};
+
+const CategoryPill = ({ name }: { name: string }) => (
+  <View style={[styles.categoryPill, categoryStyleBackgroundColor(name)]}>
+    <Text type="h3" style={[styles.categoryPillText, categoryStyleColor(name)]}>
+      {name}
+    </Text>
+  </View>
+);
+
+class CategoriesPills extends React.PureComponent<Props> {
+  static defaultProps = {
+    style: {}
+  };
+
+  scrollView: ?Object;
+
+  scrollToLastPill = () => {
+    if (!this.scrollView) {
+      return;
+    }
+
+    this.scrollView.scrollToEnd({ animated: false });
+  };
+
+  render() {
+    const { style, selectedCategories } = this.props;
+
+    return (
+      <View style={[styles.selectedCategoriesPills, style]}>
+        {selectedCategories.size === 0 ? (
+          <Text type="h3" style={styles.zeroSelected}>
+            {text.zeroSelected}
+          </Text>
+        ) : (
+          <View style={styles.scrollView}>
+            <LinearGradient
+              style={[styles.scrollShadow, styles.scrollShadowLeft]}
+              end={{ x: 0, y: 0.5 }}
+              start={{ x: 1.0, y: 0.5 }}
+              colors={["rgba(0, 0, 0, 0)", "rgba(0, 0, 0, 0.3)"]}
```

Comment:
If we start doing more gradient overlays I suggest moving these to a common place.